src/lib.rs

```diff
@@ -827,6 +833,31 @@ mod tests {
 
     use super::*;
 
+    #[test]
+    fn test_parse_correct_result() {
+        let contents = fs::read_to_string("tests/gral.bib").unwrap();
+        let option_bibliography = Bibliography::parse(&contents).unwrap();
+        let result_bibliography = Bibliography::parse_result(&contents).unwrap();
+        // Both methods should return the same bibliography for correct input files
+        assert_eq!(option_bibliography.entries, result_bibliography.entries);
+    }
+
+    #[test]
+    fn test_parse_incorrect_result() {
+        let contents = fs::read_to_string("tests/incorrect.bib").unwrap();
+
+        // Regular method should return None
+        let option_bibliography = Bibliography::parse(&contents);
+        assert!(option_bibliography.is_none());
+
+        // Result method should return the key of the incorrect entry
+        let result_bibliography = Bibliography::parse_result(&contents);
```

Comment:
... and this changed to 
```suggestion
        let result_bibliography = Bibliography::parse(&contents);
```